### PR TITLE
docs: custom ingest pipelines

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR documents custom ingest pipelines.

* An overview on how `@custom` ingest pipelines work, including limitations, has been added to the **[Parse data using ingest pipelines](https://github.com/elastic/apm-server/edit/8.3/docs/ingest-pipelines.asciidoc)** topic.
* Two tutorials will be added to the documentation. They are both linked to from the above topic
  * An APM-specific tutorial on redacting data has been added to the **[Data security](https://www.elastic.co/guide/en/apm/guide/current/apm-data-security.html)** topic.
  * A simple (add a field) Elastic integration tutorial will be added to the Elastic Agent and Fleet Guide in https://github.com/elastic/observability-docs/pull/2094.

### Related

* Closes https://github.com/elastic/observability-docs/issues/1226
* **Merge after https://github.com/elastic/observability-docs/pull/2094 as a link needs to be added to that content**

### CODE REVIEW

File: docs/ingest-pipelines.asciidoc [hunk 1]
```diff
@@ -9,23 +9,64 @@
...
-To view or edit a default pipelines in {kib},
-select **{stack-manage-app}** > **Ingest Pipelines**.
-Search for `apm`.
+The Elastic APM integration supports custom ingest pipelines.
+A custom pipeline allows you to transform data to better match your specific use case.
+This can be useful, for example, to ensure data security by removing or obfuscating sensitive information.
 
-It is not currently possible to edit or add pipelines that persist through upgrades.
+Each data stream ships with a default pipeline.
+This default pipeline calls a non-existent and non-versioned "`@custom`" ingest pipeline.
```
Comment (axw) [inline docs/ingest-pipelines.asciidoc:27]:
```suggestion
This default pipeline calls an initially non-existent and non-versioned "`@custom`" ingest pipeline.
```

File: docs/ingest-pipelines.asciidoc [hunk 2]
```diff
@@ -9,23 +9,64 @@
...
+If left uncreated, this pipeline has no effect on your data. However, if utilized,
+this pipeline can be used for custom data processing, adding fields, sanitizing data, and more.
 
-See {ref}/ingest.html[ingest node pipelines] for more information.
+In addition, ingest pipelines can also be used to direct metrics events to a data stream with a different dataset.
```
Comment (axw) [inline docs/ingest-pipelines.asciidoc:31]:
```suggestion
In addition, ingest pipelines can also be used to direct application metrics (`metrics-apm.app.*`) to a data stream with a different dataset, e.g. to combine metrics for two applications.
```

**Review by axw — APPROVED** [review]
:heart: I love the tutorial docs

File: docs/ingest-pipelines.asciidoc [hunk 3]
```diff
@@ -9,23 +9,64 @@
...
+* Create a pipeline with processors specific to your use case
+* Add the newly created pipeline to an `@custom` pipeline that matches an APM data stream
+* Roll over your data stream
+
+If you prefer more guidance, see one of these tutorials:
+
+* <<filters-ingest-pipeline>> — An APM-specific tutorial where you learn how to obfuscate passwords stored in the `http.request.body.original` field.
+* _link coming soon_ — A simple Elastic integration tutorial where you learn how to add a custom field to incoming data.
+// Add links to Fleet docs
+// To be added in https://github.com/elastic/observability-docs/pull/2094
```
Comment (bmorelli25) [inline docs/ingest-pipelines.asciidoc:70]:
```suggestion
* {fleet-guide}/data-streams-pipeline-tutorial.html[Transform data with custom ingest pipelines] — A basic Elastic integration tutorial where you learn how to add a custom field to incoming data.
```

**Pull request merged**